src/prefect/deployments/deployments.py

```diff
@@ -575,6 +575,19 @@ def validate_automation_names(cls, field_value, values, field, config):
 
         return field_value
 
+    @validator("schedule")
+    def validate_schedule(cls, value):
+        """We do not support COUNT-based (# of occurrences) RRule schedules for deployments."""
+        if value:
+            rrule_value = getattr(value, "rrule", None)
+            if rrule_value and "COUNT" in rrule_value.upper():
+                raise ValueError(
+                    "RRule schedules with `COUNT` are not supported. Please use 1."
+                    " `UNTIL`, or 2. schedule runs with the"
+                    " `/deployments/{id}/schedule` endpoint."
+                )
+        return value
```

Comment:
another option is we could potentially warn instead? but not sure how that would help.

Reply:
Raising seems like the right approach to me, a warning can be missed and confusing, whereas there's nothing ambiguous about failure.